appengine/genetics/js/cage.js

```diff
@@ -0,0 +1,730 @@
+ * @private
+ */
+Genetics.Cage.doneCallback_ = null;
+
+
+/**
+ * Stop and reset the cage.
+ */
+Genetics.Cage.reset = function() {
+  clearTimeout(Genetics.Cage.pid);
+  Genetics.Cage.EVENTS.length = 0;
+  Genetics.Cage.aliveMice_.length = 0;
+  Genetics.Cage.miceMap_ = {};
+  // Cancel all queued life simulations.
+  for (var mouseId in Genetics.Cage.mouseLife_) {
+    if (Genetics.Cage.mouseLife_.hasOwnProperty(mouseId)) {
+      clearTimeout(Genetics.Cage.mouseLife_[mouseId]);
+    }
+  }
+  Genetics.Cage.mouseLife_ = {};
+  Genetics.Cage.mouseId_ = 0;
+};
+
+/**
+ * Adds a player to the game.
+ * @param {string} playerName The name of the player to be added.
+ * @param {string|!Function} code Player's code, or generator.
+ */
+Genetics.Cage.addPlayer = function(playerName, code) {
+  // Assign a unique id to the player and add to the game.
+  var id = Genetics.Cage.playerId_++;
+  Genetics.Cage.PLAYERS[id] = [playerName, code];
+};
+
+/**
+ * Start the Cage simulation. Players should be already added
+ * @param {Function} doneCallback Function to call when game ends.
+ */
+Genetics.Cage.start = function(doneCallback) {
+  Genetics.Cage.doneCallback_ = doneCallback;
+  // Create a mouse for each player.
+  for (var playerId in Genetics.Cage.PLAYERS) {
+    if (Genetics.Cage.PLAYERS.hasOwnProperty(playerId)) {
+      for (var i = 0; i < Genetics.Cage.START_MICE_PER_PLAYER; i++) {
+        var mouseSex = i % 2 == 0 ? Genetics.Mouse.Sex.MALE :
+            Genetics.Mouse.Sex.FEMALE;
+        var mouseId = Genetics.Cage.mouseId_++;
+        var mouse = new Genetics.Mouse(mouseId, null, null, mouseSex, playerId);
+        Genetics.Cage.born(mouse);
+        new Genetics.Cage.Event('ADD', mouse,
+            Genetics.Cage.PLAYERS[playerId][0]).addToQueue();
+      }
+    }
+  }
+  Genetics.Cage.endTime_ = Date.now() + Genetics.Cage.GAME_TIME_LIMIT;
+  new Genetics.Cage.Event('START_GAME').addToQueue();
+  console.log('Starting game with ' +
+      Object.keys(Genetics.Cage.PLAYERS).length + ' players.');
+};
+
+/**
+ * Queues life simulation for mouse.
+ * @param {Genetics.Mouse} mouse The mouse to queue the simulation for.
+ */
+Genetics.Cage.life = function(mouse) {
+  // Queue life simulation for mouse with some variation in timeout to allow for
+  // variation in order.
+  Genetics.Cage.mouseLife_[mouse.id] =
+      setTimeout(goog.partial(Genetics.Cage.simulateLife, mouse),
+          10 * Math.random());
+};
+
+/**
+ * Runs life simulation on mouse.
+ * @param {Genetics.Mouse} mouse The mouse to run the simulation for.
+ */
+Genetics.Cage.simulateLife = function(mouse) {
+  // Remove life from mapping of lives that are queued to run.
+  delete Genetics.Cage.mouseLife_[mouse.id];
+  // Age mouse.
+  mouse.age++;
+  // If mouse has fight left in it, start a fight with another mouse.
+  if (mouse.aggressiveness > 0) {
+    Genetics.Cage.instigateFight(mouse);
+  }
+  // If mouse is still fertile, try to breed with another mice.
```

Comment:
Suggest moving this comment inside the condition, and rephrasing as a statement:
// Mouse is still fertile, try to breed with another mice.
Then the 'else if' can be on the same line as the closing brace.
Three places: lines 247, 251, 255.